core/moderation.go

```diff
@@ -178,6 +178,76 @@ func (n *OpenBazaarNode) GetModeratorFee(transactionTotal uint64, paymentCoin, c
 	}
 }
 
+// SetCurrencyOnListings - set currencies accepted for a listing
+func (n *OpenBazaarNode) SetCurrencyOnListings(currencies []string) error {
+	absPath, err := filepath.Abs(path.Join(n.RepoPath, "root", "listings"))
+	if err != nil {
+		return err
+	}
+
+	walkpath := func(p string, f os.FileInfo, err error) error {
+		if !f.IsDir() && filepath.Ext(p) == ".json" {
+			file, err := ioutil.ReadFile(p)
+			if err != nil {
+				return err
+			}
+			sl := new(pb.SignedListing)
+			err = jsonpb.UnmarshalString(string(file), sl)
+			if err != nil {
+				return err
+			}
+
+			sl.Listing.Metadata.AcceptedCurrencies = currencies
+			n.UpdateListing(sl.Listing)
+
+			return nil
+		}
+		return nil
+	}
+
+	err = filepath.Walk(absPath, walkpath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SetCouponOnListings - set coupons for listings
+func (n *OpenBazaarNode) SetCouponsOnListings(coupons []*pb.Listing_Coupon) error {
+	absPath, err := filepath.Abs(path.Join(n.RepoPath, "root", "listings"))
+	if err != nil {
+		return err
+	}
+
+	walkpath := func(p string, f os.FileInfo, err error) error {
+		if !f.IsDir() && filepath.Ext(p) == ".json" {
+			file, err := ioutil.ReadFile(p)
+			if err != nil {
+				return err
+			}
+			sl := new(pb.SignedListing)
+			err = jsonpb.UnmarshalString(string(file), sl)
+			if err != nil {
+				return err
+			}
+
+			sl.Listing.Coupons = coupons
```

Comment:
Should we remove the prior coupons when we apply the new ones? Merge them?

Reply:
I think it's propbably OK to replace the existing coupons as the intent is to set all the listings to the same thing. Or so it seems to me.

Reply:
That's what I was thinking, but either way we have to be clear about it and make it obvious to users what's going on.